analyses/cms-open-data-ttbar/ml_helpers.cpp

```diff
@@ -0,0 +1,74 @@
+#include "fastforest.h"
+#include <cmath>
+#include <assert.h>
+#include <map>
+#include <algorithm>
+#include "ROOT/RVec.hxx"
+
+std::map<std::string, std::vector<int>> get_permutations (std::string jet_labels) {
```

Comment:
`const std::string &` is best practice

Reply:
this is impossible, do you think so?

Reply:
in the body of function, I call `std::next_permutation(jet_labels.begin(), jet_labels.end())` which modifies jet_labels

Reply:
ah I see! then you can pick between adding a comment right above the function signature that says `copying jet_labels because we need to modify it` or take it by `const std::string &` anyway and then make a copy inside the function. either thing works: the reader won't be surprised anymore.

Reply:
added